webpack/assets/javascripts/react_app/components/common/forms/ForemanForm/integration.test.js

```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+import { IntegrationTestHelper } from '@theforeman/test';
+
+import { onError } from '../../../../redux/actions/common/forms';
+
+import { initialValues, FormComponent } from './ForemanForm.fixtures';
+
+const invalidSubmit = () =>
+  new Promise(() =>
+    onError({
+      response: {
+        status: 422,
+        data: {
+          error: {
+            errors: {
+              name: ['is too long', 'should not contain numbers'],
+              base: [
+                'does not have enough vitamins',
+                'does not have enough proteins',
+              ],
+            },
+            severity: 'warning',
+          },
+        },
+      },
+    })
+  );
+
+const props = {
+  submitForm: invalidSubmit,
+  onCancel: () => {},
+  initValues: initialValues,
+};
+
+describe('ForemanForm integration test', () => {
+  it('should render form with errors', async () => {
+    const testHelper = new IntegrationTestHelper({});
+
+    const component = testHelper.mount(<FormComponent {...props} />);
+
+    const submitBtn = component.find('Button[bsStyle="primary"]');
+    submitBtn.simulate('submit');
+    await IntegrationTestHelper.flushAllPromises();
+    component.update();
+    expect(component).toMatchSnapshot();
```

Comment:
The snapshot result looks unreasonably big, can you do some individual comparisons instead?

Reply:
I now check that form gets the expected error props and that there is an alert banner instead of having a snapshot of the whole form.